fggs/domains.py

```diff
@@ -55,3 +55,33 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not self.__eq__(other)
+
+
+class CompactDomain(Domain):
+    """A compact domain for finite sets (like vocabularies).
+
+    Compact domain is like finite domain, but it doesn't hold
+    vocabularies. Instead it only holds the size of values.
+
+    """
+
+    def __init__(self, size):
+        super().__init__()
+        self._size = size
+
+    def size(self):
+        "Return the size of the domain."
+        return self._size
+
+    def contains(self, value):
+        pass
```

Comment:
This needs to be implemented

Reply:
> This needs to be implemented

Yes. Can `value` be an `int`, or is the `Domain` interface such that it must be a `str`?

Reply:
The only place that I can imagine it would matter is in formats.py. We should modify `fgg_to_json` to handle `CompactDomain`.

Reply:
> We should modify `fgg_to_json` to handle `CompactDomain`.

I added an abstract method `to_json` to `Domain`.